src/ott/initializers/linear/initializers.py

```diff
@@ -313,3 +332,108 @@ def body_fn(i: int, f: jnp.ndarray) -> jnp.ndarray:
     return f.at[i].set(new_f)
 
   return jax.lax.fori_loop(0, len(f), body_fn, f)
+
+
+@functools.partial(jax.jit, static_argnums=(3,))
+def subsample(
+    rng: jax.random.PRNGKey, array: jnp.ndarray, weight: jnp.ndarray,
+    subsample_n: int
+):
+  """Subsample point cloud to uniform weighted points.
+
+  Args:
+    rng: random number generator.
+    array: point cloud, array of size N x d.
+    weight: array of size N.
+    subsample_n: number of points to subsample.
+
+  Returns:
+    subsampled point cloud, array of size subsample_n x d.
+  """
+  N = array.shape[0]
+  indices = jax.random.choice(
+      key=rng, a=N, shape=(subsample_n,), replace=True, p=weight, axis=0
```

Comment:
Why `replace=True`? If it's `False` and `subsample_n > N`, it will raise an error by default.

Reply:
I want to essentially re-sample the point cloud with corresponding prob=weights, to create a smaller point cloud close to the original one. This requires with replacement. 

Imagine a point cloud with 1 atom of weight 0.99 and 10^6 others with very low weight. Resampling to size 1_000 should give 1_000 points all at the same location as the atom with weight 0.99. If forced to sample without replacement then we would end up sampling some of the degenerate atoms. 

This would bias the subsampled point cloud distribution so that the expectations are not the same as the parent point cloud distribution.